OpenTrons/opentrons-api
# feat(app): adapter info in labware & liquid setup

closes RAUT-628

# Overview

This PR accomplishes 2 things:
1. refactors `getSlotLabwareName`, an old util name that no longer makes sense with what it is used for
2. changes protocol setup to match the new designs for adapters

# Test Plan

on the desktop app:

1. upload a protocol with adapters and liquids in it, go to the labware setup tab, see that the adapters are listed with their labware and that the adapter+labware combos are listed first
2. go to the liquid setup tab, see that the liquids mention the adapter and module info if applicable


on an odd:
1. send that protocol to the ODD, go to labware setup tab and see that the adapter+labware combos are listed together and if there is a H-S latch, the extra text regarding it is no longer there.

<img width="900" alt="Screen Shot 2023-08-27 at 9 15 21 PM" src="https://github.com/Opentrons/opentrons/assets/66035149/6319e5a4-bb79-427b-bbe7-dd07280fa584">
<img width="900" alt="Screen Shot 2023-08-27 at 9 15 42 PM" src="https://github.com/Opentrons/opentrons/assets/66035149/65932539-f1cf-4ed2-8174-5b9961925860">
<img width="984" alt="Screen Shot 2023-08-27 at 9 18 36 PM" src="https://github.com/Opentrons/opentrons/assets/66035149/ac7298ee-7767-458f-84b6-cc276a27f639">

# Changelog

- refactored `getSlotLabwareName` to `getLocationInfoNames` which now can return slotName, labware name, adapter name, and module model, wrote a test.
- created `getNestedLabwareInfo` which can return `nestedLabwareDisplayName`, `sharedSlotId`, `nestedLabwareDefinition`, and `nestedLabwareNickName` for the labware that is nested on top of the adapter, wrote a test.
- refactored `SetupLabwareList` and `ProtocolSetupLabware` to display the labware on top of adapter with the adapter item, fixed tests (NOTE: `ProtocolSetupLabware` needs to be cleaned up and refactored which will be done in a separate ticket https://opentrons.atlassian.net/browse/RAUT-631)
- refactored `SetupLiquidList` to display the adapter + module info if applicable.

# Review requests

see test plan

# Risk assessment

low
